libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -38,6 +40,14 @@ export class BookSearchComponent {
   }
 
   addBookToReadingList(book: Book) {
+    const mySnackBar = this._snackBar.open(`${book.title} added to reading list`, 'Undo');
+    mySnackBar.afterDismissed().subscribe(info => {
+      if (info.dismissedByAction === true) {
+        this.store.dispatch(removeFromReadingList({ item: {
+          bookId: book.id, ...book
+        } }));
+      }
+    });
```

Comment:
You are displaying the Snackbar simultaneously with dispatching the `addToReadingList` action. What happens if the `addToReadingList` action fails? Shouldn't you confirm this action before displaying the snackbar?

Reply:
Just committed a fix, I am subscribing to confirmedAddToReadingList before launching the Snackbar, however I am not sure if this is the best approach.